src/sonar_widget/SonarPlot.cc

```diff
@@ -161,7 +171,7 @@ void SonarPlot::drawOverlay()
             base::Angle ang = lastSonar.bearings[((lastSonar.beam_count - 1) * 1.0 / 4) * (i-1)];
             QPoint point(origin.rx() + BINS_REF_SIZE * sin(ang.rad) * scaleX, origin.ry() - BINS_REF_SIZE * cos(ang.rad) * scaleY);
             painter.drawLine(origin, point);
-            str.setNum(ang.getDeg());
+            str.setNum(roundf(ang.getDeg() * 10) / 10);
```

Comment:
I agree with you. Now I am using QString::number to format the number with a desired precision.
Done!